src/enemyWaves.js

```diff
@@ -0,0 +1,23 @@
+// TODO: not sure where this file belongs yet -- move if you have the onus
+import {times} from 'ramda'
+
+import {GAME_CONFIG} from './config'
+import {Enemy} from './models/enemy'
+
+const getInboundsSpawnPosition = () => {
+  const x = Math.random() * (GAME_CONFIG.width * 0.8) + GAME_CONFIG.width * 0.1
+  const y = Math.random() * (GAME_CONFIG.height / 2)
+  // TODO: very naive; should make sure it doesn't spawn under the player
+  //       or on a bullet or on another enemy etc.
+  //       or maybe better, we can have some predefined pools ala isaac
+  return {x, y}
+}
+
+export const wave = (n) => {
+  const numHard = Math.floor(n / 3)
+  const numEasy = n - numHard
+  const targetters = times(() => Enemy.TARGETTER(getInboundsSpawnPosition()), numHard)
+  const randomShooter = times(() => Enemy.PULSER(getInboundsSpawnPosition()), numEasy)
+  console.debug(`wave ${n} easy: ${randomShooter.length}, hard: ${targetters.length}`)
```

Comment:
Sorry, I think I maybe was unclear in the last review. I meant using something like [debug](https://github.com/visionmedia/debug):
```js
import bug from 'debug'
const debug = bug(`throwback:enemy:waves`)
debug(`wooooo`)
```
the `wooooo` log is only visible if the `DEBUG` environment variable is present via `process.env` in node or `localStorage` in the browser

Reply:
and by scoping our debug messages:
```js
const [npcLog, bulletLog, logLog, dogLog] = map(debug, [
  `throwback:npc`,
  `throwback:bullet`,
  `throwback:ephemera`,
  `throwback:npc:dog`,
])
```
We can print specific stuff when debugging: `DEBUG=throwback:npc:* startCommand`

Reply:
Gotcha; can save for later -- console.debug is hidden by default in chrome debug tools too